test/sharness/Makefile

```diff
@@ -14,6 +14,9 @@ IPFS_ROOT = ../..
 # User might want to override those on the command line
 GOFLAGS =
 
+# The targets in this makefile depend on being run sequentially
+.NOTPARALLEL:
```

Comment:
We may get away with adding explicit deps in the makefile. I wasn't sure if the tests themselves could be run in parallell. The main problem that spawned this is that with `make -j4` eg, the tests will start running before the clean and the deps targets have been run, which resulted in the missing ipfs binary error. Let me try that real quick.